pkg/ccl/changefeedccl/changefeed_processors.go

```diff
@@ -1467,21 +1459,8 @@ func (cf *changeFrontier) close() {
 	}
 }
 
-// closeMetrics de-registers from the progress registry that powers
-// `changefeed.max_behind_nanos`. This method is idempotent.
 func (cf *changeFrontier) closeMetrics() {
-	// Delete this feed from the MaxBehindNanos metric so it's no longer
-	// considered by the gauge.
-	func() {
-		cf.metrics.mu.Lock()
-		defer cf.metrics.mu.Unlock()
-		if cf.metricsID > 0 {
```

Comment:
It seems to me that without this check this function will no longer be idempotent which seems dangerous. Open to suggestions about what to do here. Maybe this is reason to keep cf.metrics.mu around?